CoreFeatures/Base/SimpleStrictStorageStrategy/src/main/java/info/smart_tools/smartactors/base/simple_strict_storage_strategy/SimpleStrictStorageStrategy.java

```diff
@@ -0,0 +1,79 @@
+package info.smart_tools.smartactors.base.simple_strict_storage_strategy;
+
+import info.smart_tools.smartactors.base.interfaces.i_addition_dependency_strategy.IAdditionDependencyStrategy;
+import info.smart_tools.smartactors.base.interfaces.i_addition_dependency_strategy.exception.AdditionDependencyStrategyException;
+import info.smart_tools.smartactors.base.interfaces.iresolve_dependency_strategy.IResolveDependencyStrategy;
+import info.smart_tools.smartactors.base.interfaces.iresolve_dependency_strategy.exception.ResolveDependencyStrategyException;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Simple strategy storage that uses a map to choose a concrete strategy.
+ *
+ * <p>
+ *  This strategy will throw when there is no strategy for required key that's why it is called "strict".
+ * </p>
+ */
+public class SimpleStrictStorageStrategy implements IResolveDependencyStrategy, IAdditionDependencyStrategy {
+    private final Map<Object, IResolveDependencyStrategy> strategyMap;
+    private final String resultName;
+
+    /**
+     * The constructor.
+     *
+     * @param strategyMap map to store strategies
+     * @param resultName  string that describes a resulting objects type (used for exception messages)
+     */
+    public SimpleStrictStorageStrategy(
+            final Map<Object, IResolveDependencyStrategy> strategyMap,
+            final String resultName) {
+        this.strategyMap = strategyMap;
+        this.resultName = resultName;
+    }
+
+    /**
+     * The constructor.
+     *
+     * @param resultName string that describes a resulting objects type (used for exception messages)
+     */
+    public SimpleStrictStorageStrategy(final String resultName) {
+        this(new ConcurrentHashMap<>(), resultName);
+    }
+
+    @Override
+    public void register(final Object key, final IResolveDependencyStrategy value)
+            throws AdditionDependencyStrategyException {
+        if (null == value) {
+            throw new AdditionDependencyStrategyException("Strategy should not be null.");
+        }
+
+        strategyMap.put(key, value);
+    }
+
+    @Override
+    public void remove(final Object key)
+            throws AdditionDependencyStrategyException {
+        strategyMap.remove(key);
+    }
+
+    @Override
+    public <T> T resolve(final Object... args)
+            throws ResolveDependencyStrategyException {
+        if (args.length < 1) {
+            throw new ResolveDependencyStrategyException("No arguments provided.");
+        }
+
+        IResolveDependencyStrategy strategy = strategyMap.get(args[0]);
+
+        if (null == strategy) {
+            throw new ResolveDependencyStrategyException("No " + resultName + " resolution strategy found for key '" + args[0] + "'.");
+        }
+
+        try {
+            return strategy.resolve(args);
```

Comment:
Yes, this strategy could be implemented as subclass of `StrategyStorageStrategy` but I decided to implement it from scratch as it will take the same amount of code, will have no overhead caused by callbacks and will provide a way to change `Map` implementation used.

*I don't even consider an idea of use of `StrategyStorageStrategy` directly as it makes a resulting code completely unreadable and causes lots of copy-paste*

Reply:
I've run some permanence tests for both strategies and for 1 million iteration:
And results are interesting. The StrategyStorageStrategy is not slower then SimpleStrictStorageStrategy. 
And if you want to use a specific strategy for your purposes you can create it in a specific feature otherwise we'll have the big trash in the base feature.

Reply:
The case `SimpleStrictStorageStrategy` was used in seemed to me to be very common that's why I placed it in Base feature -- to avoid a copy-pastes in future.

Reply:
But the StrategyStorageStrategy more general then SimpleStrictStorageStrategy. And the the StrategyStorageStrategy is not slower. Why should we have a many strategies for same cases?

Reply:
Because how it is done using `SimpleStrictStorageStrategy`:
```Java
storage = new SimpleStrictStorageStrategy("my things");
```

and how it's done using `StrategyStorageStrategy`:
```Java
storage = new StrategyStorageStrategy(
     // WTF
     x -> x,
     (map, key) -> {
         // WTF
         Object r = ((Map) map).get(key);
         // The check I may forget about and suddenly get NPE somwhere
         // in a code not even related this strategy definition
         if (null == r) {
             // The place where I may make a mistake copy-pasting code and thus get
             // misguiding error messages at runtime
             throw new FunctionExecutionException("No my thing named '" + key + "' found.");
         }
         return r;
     }
);
```

Reply:
For your case you can use following code:
```Java
storage = new StrategyStorageStrategy((a) -> a, (c, a) -> ((Map)c).get(a)).
```
I think it's not a big pay for universality.
And I ask you don't use a words like a 'wtf' because this repository is not a private.

Reply:
`SimpleStrictStorageStrategy` does a bit more than your code -- it also ensures that there **is** a strategy registered for the required key. The situation when there is no strategy and no exception is thrown and just a `null` is returned causes real (unlike inner classes) troubles while debugging code that uses such strategy.

When `StrategyStorageStrategy` is used with null-checks the code becomes unreadable.

Reply:
What about this:
```Java
IResolveDependencyStrategy s = new StrategyStorageStrategy(
                (a) -> a, (c, a) -> {
                        IResolveDependencyStrategy ss = ((Map<Object, IResolveDependencyStrategy>)c).get(a);
                        if (null == ss) throw new RuntimeException();
                        return ss;
                }
        );
```
This code will do what you want. And also we can add new constructor to the existing strategy for fast use this strategy in your case.
About exception on empty resolution result: I think it's a specific behavior of strategy and it's a reason to replace your strategy from base feature.
But instead we can implement two constructors (with exception on empty resolution and without exception).

Reply:
This code is still too long and contains unchecked casts. And uninformative exception. And it will be copy-pasted.

Constructors are a bad idea -- this way a generic strategy will be written in awareness of some concrete cases.

Move it away from Base?
`SimpleStrictStorageStrategy` is general enough so it does not depend on anything related to endpoints and it implements a behaviour that may and will be required by other features. Creating a separate feature containing one strategy is a bad idea as this way will lead us all into the Jar-Hell. So I have no idea where else to place this strategy.